src/agent/onefuzz-supervisor/src/commands.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::auth::Secret;
+use anyhow::Result;
+use std::process::Stdio;
+use tokio::{fs, io::AsyncWriteExt, process::Command};
+
+#[cfg(target_os = "windows")]
+use std::{env, path::PathBuf};
+
+#[cfg(target_os = "linux")]
+use users::{get_current_uid, get_user_by_name, get_user_by_uid, os::unix::UserExt};
+
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct SshKeyInfo {
+    pub key: Secret<String>,
+    pub user: Option<String>,
+    set_permissions: bool,
+}
+
+#[cfg(target_os = "windows")]
+pub async fn add_ssh_key(key_info: SshKeyInfo) -> Result<()> {
+    if key_info.user.is_some() {
+        bail!("specifying a user is not supported on windows at this time");
+    }
+
+    let mut ssh_path =
+        PathBuf::from(env::var("ProgramData").unwrap_or_else(|_| "c:\\programdata".to_string()));
```

Comment:
Ideally we'd try to grab `$SystemDrive` instead of hardcoding `c:`, but still fall back on `C:`.

Reply:
When we support arbitrary locations, handle this then.